tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -290,6 +296,25 @@ public TracerSettings(IConfigurationSource source)
         /// <seealso cref="ConfigurationKeys.ServiceVersion"/>
         public string ServiceVersion { get; set; }
 
+        /// <summary>
+        /// Gets or sets the application's git repository url.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.GitRepositoryUrl"/>
+        public string GitRepositoryUrl { get; set; }
+
+        /// <summary>
+        /// Gets or sets the application's git commit hash.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.GitCommitSha"/>
+        public string GitCommitSha { get; set; }
+
+        /// <summary>
+        /// Gets a value indicating whether we should tag every telemetry event with git metadata.
+        /// Default value is <c>true</c> (enabled).
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.GitMetadataEnabled"/>
+        public bool GitMetadataEnabled { get; }
```

Comment:
I don't think these should be public. In _theory_ they would allow the customer to set the values in code but

- we've been tending towards _not_ exposing properties like this in code, as it's problematic for future-proofing etc
- AFAICT, with the current implementation, changes in these properties _wouldn't_ be honoured by Debugger/RCM because we grab the "initial" `GitMetadataTagsProvider` which is built from static content only. 

All in all, I think it makes sense to set these as `internal` for now at least. We can always revisit if needs be if we refactor `Instrumentation`

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/commit/989c9c823ebb86da47f43eeaf48a2466a04e38af